File: refactoring-practice-1/src/main/typescript/extract_method/player-results.ts
export class PlayerResults {

    public constructor(private player1Results: number, private player2Results: number, private player3Results: number) {
    }

    public getAllPlayersResult(): string {
        let builder = {
            text: ''
        };
        this.printHeader(builder);
        builder.text += "Player 1 has " + this.player1Results + " points.\n";
        builder.text += "Player 2 has " + this.player2Results + " points.\n";
        builder.text += "Player 3 has " + this.player3Results + " points.\n";
        return builder.text;
    }

    private printHeader(builder) {
        builder.text += "Game Results\n";
        builder.text += "------------\n";
    }

}

File: refactoring-practice-1/src/main/java/dojo/extract_method/PlayerResults.java
package dojo.extract_method;

public class PlayerResults {
	
	private int player1Results;
	private int player2Results;
	private int player3Results;
	
	public PlayerResults(int player1Results, int player2Results, int player3Results) {
		this.player1Results = player1Results;
		this.player2Results = player2Results;
		this.player3Results = player3Results;
	}

	public void printAllPlayersResult() {
		printHeader();
		System.out.println("Player 1 Result is " + player1Results + " points.");
		System.out.println("Player 2 Result is " + player2Results + " points.");
		System.out.println("Player 3 Result is " + player3Results + " points.");
	}
	
	private void printHeader() {
		System.out.println("Game Results");
		System.out.println("------------");
	}
	
}
